src/freenet/node/Node.java

```diff
@@ -1312,13 +1320,47 @@ public Short get() {
 
 					@Override
 					public void set(Short val) throws InvalidConfigValueException {
-						if(maxHTL < 0) throw new InvalidConfigValueException("Impossible max HTL");
+						if(val < 0) throw new InvalidConfigValueException("Impossible max HTL");
 						maxHTL = val;
 					}
 		}, false);
 
 		maxHTL = nodeConfig.getShort("maxHTL");
 
+		 class TrafficClassCallback extends StringCallback implements EnumerableOptionCallback {
+			 @Override
+			 public String get() {
+				 return trafficClass.name();
+			 }
+
+			 @Override
+			 public void set(String tcName) throws InvalidConfigValueException {
+				 try {
+					 trafficClass = TrafficClass.fromNameOrValue(tcName);
+				 } catch (IllegalArgumentException e) {
+					 throw new InvalidConfigValueException(e);
+				 }
+			 }
+
+			 @Override
+			 public String[] getPossibleValues() {
+				 ArrayList<String> array = new ArrayList<String>();
+				 for(TrafficClass tc : TrafficClass.values())
```

Comment:
Checkstyle could help on this

Le sam. 9 janv. 2016 10:45, Florent Daigniere <notifications@github.com> a
écrit :

> In src/freenet/node/Node.java
> <https://github.com/freenet/fred/pull/474#discussion_r49262508>:
>
> > +				 return trafficClass.name();
> > +			 }
> > +
> > +			 @Override
> > +			 public void set(String tcName) throws InvalidConfigValueException {
> > +				 try {
> > +					 trafficClass = TrafficClass.fromNameOrValue(tcName);
> > +				 } catch (IllegalArgumentException e) {
> > +					 throw new InvalidConfigValueException(e);
> > +				 }
> > +			 }
> > +
> > +			 @Override
> > +			 public String[] getPossibleValues() {
> > +				 ArrayList<String> array = new ArrayList<String>();
> > +				 for(TrafficClass tc : TrafficClass.values())
>
> there has to be a better way to do this... a git-pre-commit hook? a
> stylecheck thingy that travis would run?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freenet/fred/pull/474/files#r49262508>.
>